parsing/g_vernac.ml4

```diff
@@ -243,16 +243,22 @@ GEXTEND Gram
   (* Simple definitions *)
   def_body:
     [ [ bl = binders; ":="; red = reduce; c = lconstr ->
-      let (bl, c) = expand_pattern_binders mkCLambdaN bl c in
-      (match c with
-          CCast(_,c, CastConv t) -> DefineBody (bl, red, c, Some t)
+      if List.exists (function LocalPattern _ -> true | _ -> false) bl
+      then
+        (* FIXME: "red" will be applied to types in bl and Cast with remain *)
```

Comment:
This is a bug of `'pat`. See the examples in the log of 45a4113f:
```coq
Definition f '((x,y):id nat * id nat) '((x',y'):id nat * id nat) := Eval unfold id in x+y = x'+y'.
Definition f '(x,y) '(x',y') := x+y = x'+y' : Prop.
```
We should get rid of `expand_pattern_binders`, treating `LocalPattern` as a context constructor which is interpreted in `constrintern.ml`, but this would require some work.